pydantic/main.py

```diff
@@ -1096,14 +1096,14 @@ def _calculate_keys(self, *args: Any, **kwargs: Any) -> Any:
 
 class RootModel(BaseModel, typing.Generic[RootModelRootType]):
     __pydantic_root_model__ = True
+    __pydantic_fields_set__ = {'root'}  # It's fine having a set here as it will never change
```

Comment:
Actually I think if we wanted it to be consistent with how it works for regular basemodel, this should depend on whether the default value was used. (If it needed to use the default, the value would be `set()`)

Reply:
Could you change the comment on the line so that it isn't in conflict with the TODO comment above it? Just thinking it will be a bit confusing to come back to

Reply:
Also, I think we _should_ fix this in the PR addressing the default setting for RootModel.

Reply:
It has been fixed already in https://github.com/pydantic/pydantic/pull/5949